### PR TITLE
Updates the CLI section of the SD document

### PR DESCRIPTION
**Pull request by insightmind**
This updates the CLI section of the SD document

### CODE REVIEW

File: documents/sd/chapter/Classes/Core/CLI/CLI.tex [hunk 1]
```diff
@@ -15,33 +13,305 @@ \subsection{CLI}
...
+	\end{methods}
+\end{interface}
+
+\begin{class}{InteractiveCommandline}
+	\clsdiagram[width=0.7\textwidth]{resources/Classes/Core/CLI/InteractiveCommandline.png}
+	\clsdcl{public class InteractiveCommandLine}\\
+	\clsdsp{A MessageLoop instantiates a cancelable Win32 message loop. Make sure to start the loop only on the main thread! Keep in mind that the message loop does block its calling thread, so you have to cancel it via a separate thread!}
+	
+	\begin{attributes}
+		\attribute{public bool IsRunning \{ get; private set; \} }{\\ Describes whether the MessageLoop is running or not. }
```
Comment (sjend) [inline documents/sd/chapter/Classes/Core/CLI/CLI.tex]: This doesn't seem right - perhaps Copy/Paste mistake?

File: documents/sd/chapter/Classes/Core/CLI/CLI.tex [hunk 2]
```diff
@@ -15,33 +13,295 @@ \subsection{CLI}
...
+				\para{ValidateOptions options}{\\ The resolved options used to customize the command. }
+			\end{parameters}
 		\end{method}
 	\end{methods}
+\end{class}
+
+\begin{class}{ValidateOptions}
+	\clsdiagram[width=0.3\textwidth]{resources/Classes/Core/CLI/ValidateOptions.png}
+	\clsdcl{public class ValidateOptions : CommandOptions}\\
+	\clsdsp{The ValidateOptions are to be met to executed the ValidateCommand. }
```
Comment (Gordin508) [inline documents/sd/chapter/Classes/Core/CLI/CLI.tex]:
```suggestion
	\clsdsp{The ValidateOptions are to be met to execute the ValidateCommand. }
```

File: documents/sd/chapter/Classes/Core/CLI/CLI.tex [hunk 3]
```diff
@@ -15,33 +13,295 @@ \subsection{CLI}
...
+				\para{Process options}{\\ The resolved options used to customize the command. }
+			\end{parameters}
+		\end{method}
+	\end{methods}
+\end{class}
+
+\begin{class}{ProcessOptions}
+	\clsdiagram[width=0.3\textwidth]{resources/Classes/Core/CLI/ProcessOptions.png}
+	\clsdcl{public class ProcessOptions : CommandOptions}\\
+	\clsdsp{The ProcessOptions are to be met to executed the ProcessCommand. }
```
Comment (Gordin508) [inline documents/sd/chapter/Classes/Core/CLI/CLI.tex]:
```suggestion
	\clsdsp{The ProcessOptions are to be met to execute the ProcessCommand. }
```

File: documents/sd/chapter/Classes/Core/CLI/CLI.tex [hunk 4]
```diff
@@ -15,33 +13,295 @@ \subsection{CLI}
...
+				\para{Process options}{\\ The resolved options used to customize the command. }
+			\end{parameters}
+		\end{method}
+	\end{methods}
+\end{class}
+
+\begin{class}{RecordOptions}
+	\clsdiagram[width=0.3\textwidth]{resources/Classes/Core/CLI/RecordOptions.png}
+	\clsdcl{public class RecordOptions : CommandOptions}\\
+	\clsdsp{The RecordOptions are to be met to executed the RecordCommand. }
```
Comment (Gordin508) [inline documents/sd/chapter/Classes/Core/CLI/CLI.tex]:
```suggestion
	\clsdsp{The RecordOptions are to be met to execute the RecordCommand. }
```

**Pull request merged**